controllers/admin/AdminBlockListingController.php

```diff
@@ -127,45 +131,40 @@ public function ajaxProcessSaveBlockContent()
                 $blockPsr->icon = '';
             } else {
                 $errors[] = $validUpload;
-                $this->ajaxDie(json_encode($errors));
             }
         }
-        $blockPsr->update();
+        if (empty($errors)) {
+            $blockPsr->update();
+        }
 
-        $this->ajaxDie(json_encode('success'));
+        $this->ajaxRender(json_encode(empty($errors) ? 'success' : 'error'));
     }
 
     /**
      * Reorder the blocks positions
-     *
-     * @return bool
      */
-    public function ajaxProcessUpdatePosition()
+    public function displayAjaxUpdatePosition()
     {
         $blocks = Tools::getValue('blocks');
+        $result = false;
 
-        if (empty($blocks)) {
-            return false;
-        }
-
-        foreach ($blocks as $key => $id_block) {
-            // Set the position of the Reassurance block
-            $position = $key + 1;
-
-            $dataToUpdate = [
-                'position' => (int)$position,
-            ];
-
-            $whereCondition = 'id_psreassurance = ' . (int)$id_block;
+        if (!empty($blocks)) {
```

Comment:
Watch out, because of `Tools::getValue('blocks');` This one can not be an array.
```suggestion
        if (!empty($blocks) && is_array($blocks)) {
```

Reply:
Done :)